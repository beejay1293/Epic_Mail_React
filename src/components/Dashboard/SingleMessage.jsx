import React from 'react';

import propTypes from 'prop-types';

const SingleMessage = ({
  style, name, messages, time, click, click2, del
}) => (

  <li className={style} >
    <input type="checkbox" className="checkbox" />
    <h1 className="name" onClick={click}>{name}</h1>
    <h1 className="message" onClick={click}>{messages}</h1>
    <h1 className="time" onClick={click}>{time}</h1>
    <div className="delete__icon" onClick={del}>
      <i className="fas fa-trash-alt" />
    </div>
  </li>
);

SingleMessage.propTypes = {
  style: propTypes.string.isRequired,
  name: propTypes.string.isRequired,
  messages: propTypes.string.isRequired,
  time: propTypes.string.isRequired,
};
export default SingleMessage;
