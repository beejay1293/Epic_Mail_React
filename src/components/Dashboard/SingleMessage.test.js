import React from 'react';
import { mount, shallow, render } from 'enzyme';
import SingleMessage from './SingleMessage';
describe('Single Message', () => {
  it('should render correctly a Single Message component', () => {
    const Message = shallow(<SingleMessage style='inbox__message' name='draft' messages='i need a message' time='12th' />);
  
    expect(Message).toMatchSnapshot();
  });
});
