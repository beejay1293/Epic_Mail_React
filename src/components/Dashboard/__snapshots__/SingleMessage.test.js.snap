// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Message should render correctly a Single Message component 1`] = `
<li
  className="inbox__message"
>
  <h1
    className="name"
  >
    draft
  </h1>
  <h1
    className="message"
  >
    i need a message
  </h1>
  <h1
    className="time"
  >
    12th
  </h1>
  <div
    className="delete__icon"
  >
    <i
      className="fas fa-trash-alt"
    />
  </div>
</li>
`;
